include/ncengine/ecs/Component.h

```diff
@@ -87,13 +88,22 @@ struct DefaultStoragePolicy
 template<PooledComponent T>
 struct StoragePolicy : DefaultStoragePolicy {};
 
+struct SerializationContext;
+struct DeserializationContext;
+
 /** @brief Optional data and callbacks for generic component operations. */
 template<PooledComponent T>
 struct ComponentHandler
 {
     /** @brief Function type for the Factory handler. */
     using Factory_t = std::function<T(Entity, void*)>;
 
+    /** @brief Function type for the serialize handler. */
+    using Serialize_t = std::function<void(std::ostream&, const T&, const SerializationContext&, void*)>;
```

Comment:
Nice to have param tags in the comment for consistency here.

Reply:
Yeah, the `void*` in particular is probably a little mysterious.